yt/data_objects/tests/test_covering_grid.py

```diff
@@ -109,7 +109,9 @@ def _run_tests(cg):
         assert_equal(xarr.z, cg["z"][0, 0, :])
 
     for level in [0, 1, 2]:
-        ds = fake_random_ds(16, fields=["density", "temperature", "thermal_energy"])
+        ds = fake_random_ds(
+            16, fields=["density", "temperature", "specific_thermal_energy"]
+        )
```

Comment:
Conflict with #3088 ahead! The fix is something like (check the units of the specific thermal energy)
```suggestion
        fields = ["density", "temperature", "specific_thermal_energy"]
        units = ["g/cm*3", "K", "erg/g"]
        ds = fake_random_ds(16, fields=fields, units=units)
```